Tests/HStackTests.swift

```diff
@@ -1,174 +1,764 @@
-        let view2 = UIView()
-        let size2 = CGSize(width: 55, height: 11)
-        let view3 = UIView()
-        let size3 = CGSize(width: 60, height: 12)
-        
-        let stack = HStack(spacing: spacing, thingsToStack: [
-            view1.fixed(size: size1),
-            view2.fixed(size: size2),
-            view3.fixed(size: size3)
-        ])
+final class HStackTests: XCTestCase {
+    func test_init_should_set_default_properties() {
+        let stack = HStack(
+            thingsToStack: []
+        )
+        
+        XCTAssertEqual(
+            stack.spacing,
+            0.0
+        )
+        XCTAssertEqual(
+            stack.distribution,
+            .fillEqually
+        )
+        XCTAssertEqual(
+            stack.layoutMargins,
+            .zero
+        )
+        XCTAssertEqual(
+            stack.thingsToStack.count,
+            0
+        )
+        XCTAssertNil(
+            stack.width
+        )
+    }
+    
+    func test_init_should_set_properties() {
+        let stack = HStack(
+            spacing: 8,
+            distribution: .fill,
+            layoutMargins: .init(
+                top: 10,
+                left: 10,
+                bottom: 10,
+                right: 10
+            ),
+            thingsToStack: [
+                UILabel()
+            ],
+            width: 100
+        )
+        
+        XCTAssertEqual(
+            stack.spacing,
+            8
+        )
+        XCTAssertEqual(
+            stack.distribution,
+            .fill
+        )
+        XCTAssertEqual(
+            stack.layoutMargins,
+            .init(
+                top: 10,
+                left: 10,
+                bottom: 10,
+                right: 10
+            )
+        )
+        XCTAssertEqual(
+            stack.width,
+            100
+        )
+        XCTAssertEqual(
+            stack.thingsToStack.count,
+            1
+        )
+    }
+    
+    func test_convenience_init_should_set_properties() {
+        let stack = HStack(
+            spacing: 8,
+            distribution: .fill,
+            layoutMargins: .init(
+                top: 10,
+                left: 10,
+                bottom: 10,
+                right: 10
+            ),
+            width: 100
+        ) {
+            [
+                UILabel()
+            ]
+        }
+        
+        XCTAssertEqual(
+            stack.spacing,
+            8
+        )
+        XCTAssertEqual(
+            stack.distribution,
+            .fill
+        )
+        XCTAssertEqual(
+            stack.layoutMargins,
+            .init(
+                top: 10,
+                left: 10,
+                bottom: 10,
+                right: 10
+            )
+        )
+        XCTAssertEqual(
+            stack.width,
+            100
+        )
+        XCTAssertEqual(
+            stack.thingsToStack.count,
+            1
+        )
+    }
+    
+    func test_framesForLayout_should_return_expected_frames() {
```

Comment:
It would be worth adding in the what we are testing ... in this instance `test_framesForLayout_when_items_all_fixed_size_should_return_expected_frames`

Reply:
Also worth including the distribution if these tests are all fillEqually